proposals/0050-asset-metadata.md

```diff
@@ -0,0 +1,297 @@
+    approval: MetadataModificationApproval
+}
+```
+
+##### `NoteMetadataModificationApproval`
+
+The `NoteMetadataModificationApproval` notifies the receiver consensus entity that it (or its interior) is allowed to modify certain metadata keys of an asset (or an asset instance).
+
+Similar to the `ModifyMetadata` instruction, to be executed, the `NoteMetadataModificationApproval` requires an origin. The consequences and questions are the same as for the `ModifyMetadata`.
+
+Safety: The origin must be trusted in the sense that it may grant such approvals.
+
+```rust
+NoteMetadataModificationApproval {
+    /// The ID of an asset (a collection, fungible or nonfungible).
+    asset_id: AssetId,
+
+    /// The ID of an asset instance. Applicable only to nonfungible assets.
+    ///
+    /// If `Some`, it is about metadata of an NFT.
+    /// Otherwise, is is about metadata of the collection.
+    instance: Option<AssetInstance>,
+
+    /// See the `MetadataModificationApproval` type.
+    ///
+    /// The approved origin is included in this instruction
+    /// since it could be an interior consensus entity on the receiver chain.
+    approval: MetadataModificationApproval,
+}
+```
+
+Where the `MetadataModificationApproval` is:
+
+```rust
+/// The metadata modification approval description.
+struct MetadataModificationApproval {
+    /// The origin that is approved to make modifications to the specified keys.
+    /// The list of the keys approved to be modified by the origin is specified in the `approved_keys` field.
+    approved_origin: MultiLocation,
+
+    /// The keys allowed to be modified by the `approved_origin`.
+    approved_keys: MetadataKeys,
+}
+```
+
+## Security considerations
+
+Some of the security considerations were highlighted in the description of the instructions.
+Yet, more could be considered during the discussion of the RFC.
+
+## Impact
+
+The impact of this proposal is Low (however, if we consider the change to the `Response` type as a change of the `QueryResponse` instruction, then the impact is High).
+
+The RFC introduces new instructions, so XCVM implementations have to be updated.
+
+## Alternatives
+
+All the proposed instructions could be replaced with `Transact`.
+However, without the proposed instructions, there is no standard communication language about metadata in XCM. Using `Transact` for this purpose could be inconsistent and more error-prone.
+
+## Questions and open Discussions
+
+1. Naming of all the proposed instructions and types
+2. Is there a way to simplify or make the proposed instructions more convenient?
+3. Are the proposed metadata format and operations general enough?
+4. Could the proposed instruction set be reduced without losing both generality and convenience?
+5. How one could easily pay for the execution of the `ModifyMetadata`, `ApproveMetadataModification`, and `NoteMetadataModificationApproval`?
+6. How to make the `MetadataMap` bounded? The same question for `MetadataKeys`.
```

Comment:
the simply way is define max key length, max value length and max number of keys
another ways is not define anything here, but depends on the overall XCM size limit